README.md

```diff
@@ -25,6 +25,42 @@ void main() {
 }
 ```
 
+## Generation conventions
+
+The generator scripts use a number of conventions they use consistently to
+handle Web IDL definitions:
+
+### Interfaces
+
+- Interfaces are emitted as extension types that wrap and implement `JSObject`.
+- Interface inheritance is maintained using `implements`.
+- Members of partial interfaces and partial mixins are added to the interfaces
+  and mixins that include them, and therefore do not have separate declarations.
+- Members of mixins are added to the interfaces that include them, and therefore
```

Comment:
(optional) could be combined with the previous bullet point. For example: "Members of mixins, partial mixins, and partial interfaces are added to ..."